File: BOJ/#2745 진법 변환/진법 변환.cpp
#include <iostream>
using namespace std;

int main() {
    string N; int B;
    cin>>N>>B;

    int answer = 0;
    for(int i=0; i<N.length(); i++) {
        if(isdigit(N[i]))
            answer = answer*B+(N[i]-'0');
        else answer = answer*B+(N[i]-'A'+10);
    }

    cout<<answer;
}

File: BOJ/#2745 진법 변환/진법 변환.py
N,B = input().split()
B = int(B)

answer = 0
for i in range(len(N)):
    if N[i].isdigit():
        answer = answer*B+int(N[i])
    else: answer = answer*B+(ord(N[i])-ord('A')+10)

print(answer)
